### PR TITLE
Include failed HRESULTs and exceptions in FREB log

### PR DESCRIPTION
**Pull request by pakrym**
We accomplish this by setting thread local ITraceContext and use it from our HRESULT and exception helpers.

Sample result:

image

### CODE REVIEW

File: src/AspNetCoreModuleV2/AspNetCore/ApplicationFactory.h [hunk 1]
```diff
@@ -34,9 +34,10 @@ class ApplicationFactory
         _In_  IHttpContext          *pHttpContext,
         _Outptr_ IAPPLICATION       **pApplication) const
     {
+        auto location = m_location;
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/AspNetCore/ApplicationFactory.h]: Context of this change? Add a comment about why you do this if intentional.

File: src/AspNetCoreModuleV2/AspNetCore/ancm.mof [hunk 2]
```diff
@@ -306,3 +306,66 @@ class ANCMManagedRequestCompletion:ANCM_Events
      read]
      object  ContextId;
 };
+
+[Dynamic,
+ Description("Failed HRESULT") : amended,
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/AspNetCore/ancm.mof]: Can you fix the previous events such that all have unique event types?

File: src/AspNetCoreModuleV2/CommonLib/exceptions.h [hunk 3]
```diff
@@ -74,6 +77,26 @@
 #define SUCCEEDED_LOG(hr)                                       SUCCEEDED(LOG_IF_FAILED(hr))
 #define FAILED_LOG(hr)                                          FAILED(LOG_IF_FAILED(hr))
 
+inline thread_local IHttpTraceContext* g_traceContext;
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/exceptions.h]: It's weird to have this here. Can you add this to ModuleTracer. Or at least make:
```
    if (ANCMEvents::ANCM_HRESULT_FAILED::IsEnabled(pTraceContext))
    {
        ANCMEvents::ANCM_HRESULT_FAILED::RaiseEvent(pTraceContext, nullptr, fileName, lineNumber, hr);
    }
```
a static method in ModuleTracer.h

File: src/AspNetCoreModuleV2/CommonLib/requesthandler.h [hunk 4]
```diff
@@ -30,7 +38,26 @@ class REQUEST_HANDLER: public virtual IREQUEST_HANDLER
...
+        return ExecuteRequestHandler();
+    }
+
+    REQUEST_NOTIFICATION_STATUS
+    OnAsyncCompletion(
+        DWORD      cbCompletion,
+        HRESULT    hrCompletionStatus
+    ) final
+    {
+        auto traceScope = TraceContextScope(m_pHttpContext.GetTraceContext());
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/requesthandler.h]: Yeah, if you integrate ModuleTracer and TraceContextScope together, that'd be great.

File: src/AspNetCoreModuleV2/CommonLib/requesthandler.h [hunk 5]
```diff
@@ -30,7 +38,26 @@ class REQUEST_HANDLER: public virtual IREQUEST_HANDLER
         }
     }
 
-    REQUEST_NOTIFICATION_STATUS OnAsyncCompletion(DWORD cbCompletion, HRESULT hrCompletionStatus) override
+
+    REQUEST_NOTIFICATION_STATUS
+    OnExecuteRequestHandler() final
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/requesthandler.h]: I think this is safe, but let's make sure the namechange is forwards and backwards compatible.

File: src/AspNetCoreModuleV2/AspNetCore/ancm.mof [hunk 6]
```diff
@@ -5,7 +5,8 @@
 /*
  * AspNetCore module trace events layout
  * Uncomment the following class to run mof2trace to generate header file
- * comment it back before checking it in 
+ * comment it back before checking it in
+
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/AspNetCore/ancm.mof]: revert new line changes.

**Comment by jkotalik** [comment]
Fixup compilation issues with traceconfig.

**Comment by pakrym** [comment]
Some kind of Compiler bug, passed VSTS

**Pull request merged**